yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MultiServerConnectionFactory.java

```diff
@@ -64,25 +69,34 @@ public enum ConnectionSelectionPolicy { DEFAULT, ALTERNATE }
     private final List<YamlRunner.YamlConnectionFactory> alternateFactories;
 
     public MultiServerConnectionFactory(@Nonnull final YamlRunner.YamlConnectionFactory defaultFactory,
-                                        @Nullable final List<YamlRunner.YamlConnectionFactory> alternateFactories) {
+                                        @Nonnull final List<YamlRunner.YamlConnectionFactory> alternateFactories) {
```

Comment:
Do we have to change the parameter to @Nonnull? I guess we can still manage with only "current version" connection, which would translate to "current version"?
I think that even with this change, if we have two "embedded connection" factories, the result would still have an empty version set.

Reply:
I guess I should have followed through with changing downstream, but AFAICT, there is no difference between passing in `null` and `List.of()`, and if we change this to `@Nonnull`, we can remove a bunch of nullability checks below.

Reply:
Yeah, empty list will work too.